vochain/scrutinizer/scrutinizer.go

```diff
@@ -56,52 +63,103 @@ func NewScrutinizer(dbPath string, state *vochain.State) (*Scrutinizer, error) {
+	// Some grace time to avoid false positive on IsSynchronizing()
+	time.Sleep(time.Second * 10)
+	for s.VochainState.IsSynchronizing() {
+		time.Sleep(time.Second * 2)
+	}
+	log.Infof("running scrutinizer after-sync bootstrap")
+	prcs := bytes.Buffer{}
+	err := s.db.ForEach(
+		badgerhold.Where("FinalResults").Eq(false),
+		func(p *Process) error {
+			if !p.Envelope.EncryptedVotes {
+				s.addProcessToLiveResults(p.ID)
+				prcs.WriteString(fmt.Sprintf("%x ", p.ID))
+			}
+			return nil
+		})
+	if err != nil {
+		log.Error(err)
+	}
+	log.Infof("current live results processes: %s", prcs.String())
+	// TODO: COUNT ALL PENDING LIVE VOTES
+}
+
 // Commit is called by the APP when a block is confirmed and included into the chain
-func (s *Scrutinizer) Commit(height int64) {
+func (s *Scrutinizer) Commit(height int64) (error, bool) {
 	// Add Entity and register new active process
-	var nvotes int64
 	for _, p := range s.newProcessPool {
 		if err := s.newEmptyProcess(p.ProcessID); err != nil {
-			log.Warnf("commit: cannot create new empty process: %v", err)
+			log.Errorf("commit: cannot create new empty process: %v", err)
+			continue
 		}
-	}
+		if live, err := s.isOpenProcess(p.ProcessID); err != nil {
+			log.Errorf("cannot check if process is live results: %v", err)
+
+		} else if live && !s.VochainState.IsSynchronizing() {
+			// Only add live processes if the vochain is not synchronizing
+			s.addProcessToLiveResults(p.ProcessID)
 
-	// Add votes collected by onVote (live results)
-	for _, v := range s.votePool {
-		if err := s.addLiveVote(v); err != nil {
-			log.Errorf("cannot add live vote: (%s)", err)
-			continue
 		}
-		nvotes++
-	}
-	if nvotes > 0 {
-		log.Infof("added %d live votes from block %d", nvotes, height)
 	}
 
 	// Update existing processes
 	for _, p := range s.updateProcessPool {
 		if err := s.updateProcess(p); err != nil {
-			log.Warnf("commit: cannot update process %x: %v", p, err)
+			log.Errorf("commit: cannot update process %x: %v", p, err)
+			continue
 		}
 	}
+
+	// Schedule results computation
 	for _, p := range s.resultsPool {
 		if err := s.setResultsHeight(p.ProcessID, uint32(height+1)); err != nil {
-			log.Warnf("commit: cannot update process %x: %v", p.ProcessID, err)
+			log.Errorf("commit: cannot update process %x: %v", p.ProcessID, err)
+			continue
 		}
+		s.delProcessFromLiveResults(p.ProcessID)
 		log.Infof("scheduled results computation on next block for %x", p.ProcessID)
 	}
 
 	// Check if there are processes that need results computing
 	// this can be run async
 	go s.computePendingProcesses(uint32(height))
+
+	// Add votes collected by onVote (live results)
+	// this can be run async
+	go func() {
+		nvotes := 0
+		timer := time.Now()
+		for pid, votes := range s.votePool {
+			results := &Results{Weight: new(big.Int).SetUint64(0)}
+			for _, v := range votes {
+				if err := s.addLiveVote(v, results); err != nil {
+					log.Warnf("vote cannot be added: %v", err)
```

Comment:
Warning is fine since the user might provide a wrong Vote package and that should be considered as a "null vote", it's not an error and probably in the future (when we are sure everything works as expected) we can even remove the warning